### PR TITLE
Configure pagination

### PR DESCRIPTION
**Pull request by benjamineskola**
### Description of change

- **Update Pagy**
- **Match Pagy component to GOV.UK standard**
- **Handle pagination overflow without an exception** (see #1805)
- **Add pagy links to search response**

### CODE REVIEW

**Review by pixeltrix — APPROVED** [review]

**Review by benbaumann95 — APPROVED** [review]

File: engines/bops_api/app/views/bops_api/v2/public/search.json.jbuilder [hunk 1]
```diff
@@ -12,3 +12,5 @@ end
 json.data @planning_applications.each do |planning_application|
   json.partial! "show", planning_application:
 end
+
+json.links pagy_jsonapi_links(@pagy)
```
Comment (benbaumann95) [inline engines/bops_api/app/views/bops_api/v2/public/search.json.jbuilder:16]: nitpick but do we want to put this at the top of the response?

Comment (benjamineskola) [reply engines/bops_api/app/views/bops_api/v2/public/search.json.jbuilder]: I'm not sure whether it'll matter but it certainly can't hurt.

File: engines/bops_api/app/views/bops_api/v2/public/search.json.jbuilder [hunk 2]
```diff
@@ -9,6 +9,8 @@ json.metadata do
   json.total_results @pagy.count
 end
 
+json.links pagy_jsonapi_links(@pagy)
```
[the remark this answers is not in the document]
Comment (pixeltrix) [reply engines/bops_api/app/views/bops_api/v2/public/search.json.jbuilder:12]: Prefer absolute paths - relative doesn't make sense with an api

**Comment by benjamineskola** [comment]
Swagger examples added and schema validation works.

**Pull request merged**